File: NodeJS/AddTranslationArray.js
'use strict';

let https = require ('https');

// **********************************************
// *** Update or verify the following values. ***
// **********************************************

// Replace the subscriptionKey string value with your valid subscription key.
let subscriptionKey = 'ENTER KEY HERE';

let host = 'api.microsofttranslator.com';
let path = '/V2/Http.svc/AddTranslationArray';

let params = '';

let from = "en-us";
let to = "fr-fr";
let original = "Hi there";
let translation = "Salut";
let user = "JohnDoe";

let ns = "http://schemas.datacontract.org/2004/07/Microsoft.MT.Web.Service.V2";
let content =
	"<AddtranslationsRequest>" +
	"  <AppId />" +
	"  <From>" + from + "</From>" +
	"  <Options>" +
	"    <User xmlns=\"" + ns + "\">" + user + "</User>" +
	"  </Options>" +
	"  <To>" + to + "</To>" +
	"  <Translations>" +
	"    <Translation xmlns=\"" + ns + "\">" +
	"      <OriginalText>" + original + "</OriginalText>" +
	"      <Rating>1</Rating>" +
	"      <TranslatedText>" + translation + "</TranslatedText>" +
	"    </Translation>" +
	"  </Translations>" +
	"</AddtranslationsRequest>";

let response_handler = function (response) {
    let body = '';
    response.on ('data', function (d) {
        body += d;
    });
    response.on ('end', function () {
		console.log (body);
    });
    response.on ('error', function (e) {
        console.log ('Error: ' + e.message);
    });
};

let AddTranslationArray = function () {
	let request_params = {
		method : 'POST',
		hostname : host,
		path : path + params,
		headers : {
			'Content-Type' : 'text/xml',
			'Ocp-Apim-Subscription-Key' : subscriptionKey,
		}
	};

	let req = https.request (request_params, response_handler);
	req.write (content);
	req.end ();
}

AddTranslationArray ();

File: CSharp/AddTranslationArray.cs
using System;
using System.Net.Http;
using System.Text;
using System.Xml.Linq;

namespace TranslateTextQuickStart
{
    class Program
    {
        static string host = "https://api.microsofttranslator.com";
        static string path = "/V2/Http.svc/AddTranslationArray";

        // NOTE: Replace this example key with a valid subscription key.
        static string key = "ENTER KEY HERE";

        async static void AddTranslationArray()
        {
            string uri = host + path;

            XNamespace ns = @"http://schemas.datacontract.org/2004/07/Microsoft.MT.Web.Service.V2";
            XDocument doc = new XDocument(
                new XElement("AddtranslationsRequest",
                    // NOTE: AppId is required, but it can be empty because we are sending the Ocp-Apim-Subscription-Key header.
                    new XElement("AppId"),
                    new XElement("From", "en-US"),
                    new XElement("Options",
                        new XElement(ns + "User", "JohnDoe")
                    ),
                    new XElement("To", "fr-fr"),
                    new XElement("Translations",
                        new XElement(ns + "Translation",
                            new XElement(ns + "OriginalText", "Hi there"),
                            new XElement(ns + "Rating", 1),
                            new XElement(ns + "TranslatedText", "Salut")
                        )
                    )
                )
            );
            var requestBody = doc.ToString();

            using (var client = new HttpClient())
            using (var request = new HttpRequestMessage())
            {
                request.Method = HttpMethod.Post;
                request.RequestUri = new Uri(uri);
                request.Content = new StringContent(requestBody, Encoding.UTF8, "text/xml");
                request.Headers.Add("Ocp-Apim-Subscription-Key", key);
                var response = await client.SendAsync(request);
                Console.WriteLine(response.ToString());
            }
        }

        static void Main(string[] args)
        {
            AddTranslationArray();
            Console.ReadLine();
        }
    }
}
